pkg/discovery/stitching/volume_stitching_manager.go

```diff
@@ -198,16 +198,41 @@ func (aws *awsVolumeUUIDGetter) GetVolumeUUID(vol *api.PersistentVolume) (string
 		return "", fmt.Errorf("not a valid AWS provisioned volume: %v", vol.Name)
 	}
 
-	return extractAWSVolumeUuid(vol.Spec.AWSElasticBlockStore.VolumeID, vol.Name)
+	return extractAWSVolumeUuid(vol.Spec.AWSElasticBlockStore.VolumeID, vol)
 }
 
-func extractAWSVolumeUuid(volID, volName string) (string, error) {
+func extractAWSVolumeUuid(volID string, vol *api.PersistentVolume) (string, error) {
+	volName := vol.Name
 	//1. split the suffix into two parts:
 	// aws://us-east-2c/vol-0e4eaa3ef79bcb5a9 -> [us-east-2c, vol-0e4eaa3ef79bcb5a9]
-	suffix := volID[len(awsVolPrefix):]
+	suffix := volID
+	if strings.HasPrefix(volID, awsVolPrefix) {
+		suffix = volID[len(awsVolPrefix):]
+	}
+
 	parts := strings.Split(suffix, "/")
-	if len(parts) != 2 {
-		return "", fmt.Errorf("failed to split uuid (%d): %v, for volume %v", len(parts), parts, volName)
+	if len(parts) < 2 {
```

Comment:
Line 246 below can cause a problem if this condition is true.
`parts[1]` does not seem to be set (nil pointer dereference) (`len(parts) < 2`)